src/components/themed/WalletListSwipeable.tsx

```diff
@@ -93,7 +94,33 @@ function WalletListSwipeableComponent(props: Props) {
   })
 
   // Filter based on the search text:
-  const searchedWalletList = React.useMemo(() => searchWalletList(sortedWalletList, searchText), [sortedWalletList, searchText])
+  const searchedWalletList = React.useMemo(() => {
+    const list = searchWalletList(sortedWalletList, searchText)
+    if (searchText === '' || mostRecentWallets.length === 0) return list
+
+    const recentIndex: Record<string, number> = {}
+    for (let i = 0; i < mostRecentWallets.length; ++i) {
+      const { id, currencyCode } = mostRecentWallets[i]
+      recentIndex[`${id}_${currencyCode.toLowerCase()}`] = i
+    }
+
+    return list
+      .map((item, index) => ({ item, index }))
+      .sort((a, b) => {
+        if (a.item.type === 'asset' && b.item.type === 'asset') {
+          const keyA = `${a.item.wallet.id}_${(a.item.token?.currencyCode ?? a.item.wallet.currencyInfo.currencyCode).toLowerCase()}`
+          const keyB = `${b.item.wallet.id}_${(b.item.token?.currencyCode ?? b.item.wallet.currencyInfo.currencyCode).toLowerCase()}`
+
+          const pa = recentIndex[keyA]
+          const pb = recentIndex[keyB]
+          if (pa != null && pb != null) return pa - pb
+          if (pa != null) return -1
+          if (pb != null) return 1
+        }
+        return a.index - b.index
+      })
+      .map(({ item }) => item)
```

Comment:
While this works, it's a weird way to do it. I would have done something more like this:

```ts
// Make a list of recent wallets:
let recentWallets = []
for (const {id, currencyCode} of mostRecentWallets) {
  const item = list.find(...)
  if (item != null) recentWallets.push(item)
}

// Put the recent wallets in the front:
return [
  ...recentWallets,
  ...list.filter(item => !recentWallets.includes(item))
]
```

However, I do see that your way works, so this is just a suggestion. I still approve your version, even if it's a bit heavy.